spec/support/shared_examples.rb

```diff
@@ -0,0 +1,14 @@
+RSpec.shared_examples '#canonicalize' do |provider|
+  let(:resource) do
+    {
+      name: 'foo',
+      ensure: 'present',
+    }
+  end
+
+  context 'canonicalize is called' do
+    it 'returns passed in resource' do
+      expect(provider.canonicalize(anything, resource)).to eq(name: 'foo', ensure: 'present')
```

Comment:
the title and content of this example do not match: the title claims the same object instance is returned (which you can check using the [`object_id`](https://ruby-doc.org/core-2.5.1/Object.html#method-i-object_id), while the test checks that the values were not modified.

both sound like good tests.